stack/base/before-notebook.d/40_prepare-aiida.sh

```diff
@@ -74,6 +74,8 @@ load_computer('${computer_name}').set_minimum_job_poll_interval(${job_poll_inter
 else
 
   # Migration will run for the default profile.
+  ## We need to stop the daemon before.
+  verdi daemon stop
```

Comment:
I think we need to understand why this is needed, since the daemon cannot be running at the container startup.

@sphuber does `verdi daemon stop` do some kind of cleanup that might explain why it is needed here? It is true that we currently don't gracefully stop the daemon when we stop the container so there might perhaps be some stale pid files or something?

@mikibonacci what is the exact error that you saw coming from the `verdi storage migrate` command?

Reply:
Depends on the version. The behavior has changed a bit in the 2.x minor releases. But currently, yes, `verdi daemon stop` should clean up stale PID-files. And it is possible that some other commands will check if the daemon "is running" based on that pid file. Would have to look into more detail if you know exactly which version you are targeting

Reply:
Thanks! We're currently on 2.5, and hopefully will switch to 2.6 soon.

I was trying to get rid of `verdi daemon stop` because it adds 2s to container startup. Perhaps we can just remove the stale pid files manually?

Reply:
Hi @danielhollas  and @sphuber, the exact error printed in the log is:

`2024-07-03 15:41:58 Critical: Migration aborted, the daemon for the profile is still running.`

The target AiiDA version is `aiida-core==2.5.1` (build=pyhca7485f_0, channel=conda-forge) indeed.

Reply:
> I was trying to get rid of verdi daemon stop because it adds 2s to container startup. Perhaps we can just remove the stale pid files manually?

Well if you are absolutely certain that the daemon shouldn't be running, you could just manually remove the pid file `.aiida/daemon/circus-{profile-name}.pid` which should be very fast. The `verdi daemon stop` command will actually try to reach the daemon, which has a 2 second timeout by default.

Reply:
Command `verdi storage migrate` has access to daemon client, so probably another solution is when `--force` flag is used, it does clean stale PID files before migrate.

Reply:
I am not sure that it would be a good idea to have --force ignore a running daemon. The --force flag is there to not have to type out the confirmation message. This has been the case for a long time and users may have grown used to that. If we now also have it start ignoring a running daemon, that might be dangerous.

Reply:
Okay, then manually delete it as Miki did here is fine.